src/receive_imf/tests.rs

```diff
@@ -3502,8 +3512,11 @@ async fn test_dont_readd_with_normal_msg() -> Result<()> {
 
     bob.recv_msg(&alice.pop_sent_msg().await).await;
 
-    // Alice didn't receive Bobs leave message, but bob shouldn't readded himself just because of that.
-    assert!(!is_contact_in_chat(&bob, bob_chat_id, ContactId::SELF).await?);
+    // Alice didn't receive Bob's leave message, so Bob must readd themselves otherwise other
```

Comment:
I have no better idea currently how to handle this. Bob mustn't just ignore his implicit re-addition, maybe we should implement a leave auto-retry, but as Alice could re-add Bob on purpose, the logic would become too complicated and i'd better avoid such auto-retries